tests/zkevm/test_worst_compute.py

```diff
@@ -741,6 +741,101 @@ def test_worst_unop(
     )
 
 
+@pytest.mark.valid_from("Cancun")
+@pytest.mark.parametrize("key_mut", [True, False])
+@pytest.mark.parametrize("val_mut", [True, False])
+def test_worst_tload(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    key_mut: bool,
+    val_mut: bool,
+):
+    """Test running a block with as many TLOAD calls as possible."""
+    env = Environment()
+
+    start_key = 41
+    code_key_mut = Bytecode()
+    code_val_mut = Bytecode()
+    if key_mut and val_mut:
+        code_prefix = Op.PUSH1(start_key) + Op.JUMPDEST
+        code_key_mut = Op.POP + Op.GAS
+        code_val_mut = Op.TSTORE(Op.DUP2, Op.GAS)
+        loop_iter = Op.POP(Op.TLOAD(Op.DUP1))
```

Comment:
Ah ok wait this is somewhat confusing, so in this case the key is only edited after each loop, while at the `key_mut = True` `val_mut = False` the key is actually edited each loop "item"

Reply:
For readability: I would put `loop_iter` directly after `code_prefix` :thinking:

Reply:
> Ah ok wait this is somewhat confusing, so in this case the key is only edited after each loop, while at the key_mut = True val_mut = False the key is actually edited each loop "item"

No no, `code_key_mut` and `code_val_mut` are always appended in `code_suffix` -- thus both are only applied once per big-loop. (This applies to all cases)

Per iteration we use Op.DUP1 since is the place for the current key value.

Reply:
I'm talking about this:

```python
    if key_mut and val_mut:
        code_prefix = Op.PUSH1(start_key) + Op.JUMPDEST
        loop_iter = Op.POP(Op.TLOAD(Op.DUP1))
        code_key_mut = Op.POP + Op.GAS
        code_val_mut = Op.TSTORE(Op.DUP2, Op.GAS)
    if key_mut and not val_mut:
        code_prefix = Op.JUMPDEST
        loop_iter = Op.POP(Op.TLOAD(Op.GAS))
```

In the first code block, `TLOAD` is the same key for each loop "item" right (because it `TLOAD(Op.DUP1)`? In the second code block, it is different for each loop "item", however (because it `TLOAD(Op.GAS)`)?

Reply:
Ahh, I see what you mean. You're correct. I think the way it is now is the best since sounds more dynamic, but then I pushed a commit now to fix the comment in the parameters of the test (which I think was related to other comment you did).